client/state/inline-help/reducer.js

```diff
@@ -82,6 +83,11 @@ export const search = withoutPersistence(
 					selectedResult: -1,
 					hasAPIResults: false,
 				};
+			case INLINE_HELP_SET_QUERY_SEARCH:
```

Comment:
I notice that `INLINE_HELP_SET_QUERY_SEARCH` and `INLINE_HELP_SEARCH_REQUEST are identical. 

Why is this? 

Do we want to remove one or the other? It certainly feels like an issue to have two actions for the same action.

Reply:
Yes, it's confusing. It seems that we could remove one of them. I added `INLINE_HELP_SET_QUERY_SEARCH` because that is what the action does. Just set the query which will be used to define the request when it's triggered. It doesn't request anything. Let me dig a little bit on this.

Reply:
Thinking well, in general, we define three actions when the app needs to perform an async request.

* `REQUEST `: the request starts. It's useful to know if it _isRequesting_.
* `REQUEST_SUCCESS`: request ends up successfully.
* `REQUEST_FAILURE`: request ends up in an unexpected way.

So maybe we should consider keeping both actions. `INLINE_HELP_SET_QUERY_SEARCH` to set the query, and `INLINE_HELP_SEARCH_REQUEST` to start to request.

Reply:
> Just set the query which will be used to define the request when it's triggered.

Within the `search` portion of state the `INLINE_HELP_SEARCH_REQUEST` is setting the search query text. That's it. 

The "requesting" state is set in the `requesting` reducer and is completely separate from the `search` reducer. This is where the action type `INLINE_HELP_SEARCH_REQUEST` is used.

Therefore, I think we should retain the action type and use it in the `requesting` reducer, but remove it from the `search` reducer as it's not longer needed.

Reply:
> > Just set the query which will be used to define the request when it's triggered.
> 
> Within the `search` portion of state the `INLINE_HELP_SEARCH_REQUEST` is setting the search query text. That's it.

yeah, maybe that's a mistake. This action should be only used to trigger the request, imo.

> The "requesting" state is set in the requesting reducer and is completely separate from the search reducer.

Agree, yes! it's aligned with what I said before.

> Therefore, I think we should retain the action type and use it in the requesting reducer, but remove it from the search reducer as it's not longer needed.

#### In short
* The UI should take over setting the search query. In this case, the `<SearchCad />` sets the query when running the `setInlineHelpSearchQuery()` (change purposed on this PR)
* The action ``requestInlineHelpSearchResults` function which performs the request should not take care of the query but all requesting process: populate items with the endpoint response, handle errors if something fails, reset when there aren't results, etc. Need to keep in an open eye here.
* The connection between the UI and data flow is made by the data query component. 💥 
```es6
<QueryInlineHelpSearch query={ searchQuery } />
```
When the `searchQuery` changes, `<QueryInlineHelpSearch />` performs a new request and the reducer populates the state with the endpoint response.